monai/inferers/splitter.py

```diff
@@ -56,144 +57,210 @@ def __call__(self, inputs: Any) -> Iterable[tuple[torch.Tensor, Sequence[int]]]:
         """
         raise NotImplementedError(f"Subclass {self.__class__.__name__} must implement this method.")
 
+    @staticmethod
+    def _validate_patch_filter_fn(filter_fn):
+        if callable(filter_fn):
+            sig = signature(filter_fn)
+            n_params = len(sig.parameters)
+            num_pos_params = len([v for v in sig.parameters.values() if v.default is _empty])
+            if n_params < 2:
+                raise ValueError(
+                    f"`patch_filter_fn` requires to accept at least two parameters (patch, location)."
+                    f"The provided callable ({filter_fn}) has {n_params} parameters."
+                )
+            elif num_pos_params > 2:
+                raise ValueError(
+                    f"`patch_filter_fn` can have at most two positional parameters (patch, location)."
+                    f"The provided callable ({filter_fn}) has {num_pos_params} positional parameters."
+                )
+        elif filter_fn is not None:
+            raise ValueError(
+                "`patch_filter_fn` should be a callable with two input parameters (patch, location). "
+                f"{type(filter_fn)} is given."
+            )
+        return filter_fn
+
 
 class SlidingWindowSplitter(Splitter):
+    """Split the input into patches with sliding window strategy and a possible overlap.
+    It also allows to offset the starting position and filter the patches.
+
+    Args:
+        patch_size : the size of the patches to be generated.
+        offset: the amount of offset for the patches with respect to the original input.  Defaults to 0.
+        overlap: the amount of overlap between patches in each dimension [0, 1). Defaults to 0.0.
+        filter_fn: a callable to filter patches. It should accepts exactly two parameters (patch, location), and
+            return True for a patch to keep. Defaults to no filtering.
+        device: the device where the patches are generated. Defaults to the device of inputs.
+        non_spatial_ndim: number of non-spatial dimensions (e.g. batch, color)
+        pad_kwargs: arguments for `torch.nn.functional.pad`.
+            For padding the input images in order to capture the patches crossing the border of the image
+            (either when the offset is negative or the image is non-divisible by the patch_size),
+            "mode" need to be set, e.g., {"mode": "constant"}.
+        reader: the module to be used for loading whole slide imaging. If `reader` is
```

Comment:
I personally feel this `SlidingWindowSplitter` should be a general purpose class, we should not put WSI specific logic here.
I will vote to develop a subclass as your initial idea.
@wyli @ericspod What do you think?

Thanks.